import_logs.py

```diff
@@ -1527,9 +1538,10 @@ def _call(self, path, args, headers=None, url=None, data=None):
             self.RedirectHandlerWithLogging(),
             urllib.request.HTTPSHandler(**https_handler_args))
         response = opener.open(request, timeout = timeout)
+        encoding = response.info().get_content_charset('utf-8')
         result = response.read()
         response.close()
-        return result
+        return result.decode(encoding)
```

Comment:
why exactly are those change required?

Reply:
I had the same problem as this issue : https://github.com/matomo-org/matomo-log-analytics/issues/286

Here is the result when I was running on the server:

`hosting-user@server:/srv/data$ python3.5 ./web/vhosts/stats.example.com/htdocs/misc/log-analytics/import_logs.py --url=https://stats.example.com --debug --debug-tracker --token-auth=f1f1f1f1f11f1f1f1f1f1f1f1f1f1 --idsite=1 --hostname=www.example.com --enable-http-errors --enable-http-redirects --enable-static --enable-bots --log-format-name=gandi ./home/access.log
2021-06-25 08:56:50,923: [DEBUG] Accepted hostnames: www.example.comstatic --ena                                              2021-06-25 08:56:50,924: [DEBUG] Matomo Tracker API URL is: https://stats.example.com
2021-06-25 08:56:50,924: [DEBUG] Matomo Analytics API URL is: https://stats.example.com
2021-06-25 08:56:50,924: [DEBUG] Authentication token token_auth is: f1f1f1f1f11f1f1f1f1f1f1f1f1f1
2021-06-25 08:56:50,924: [DEBUG] Resolver: static
Traceback (most recent call last):
  File "./web/vhosts/stats.example.com/htdocs/misc/log-analytics/import_logs.py", line 2687, in <module>
    resolver = config.get_resolver()
  File "./web/vhosts/stats.example.com/htdocs/misc/log-analytics/import_logs.py", line 1129, in get_resolver
    return StaticResolver(self.options.site_id)
  File "./web/vhosts/stats.example.com/htdocs/misc/log-analytics/import_logs.py", line 1663, in __init__
    'SitesManager.getSiteFromId', idSite=self.site_id
  File "./web/vhosts/stats.example.com/htdocs/misc/log-analytics/import_logs.py", line 1646, in call_api
    return self._call_wrapper(self._call_api, None, None, method, **kwargs)
  File "./web/vhosts/stats.example.com/htdocs/misc/log-analytics/import_logs.py", line 1598, in _call_wrapper
    response = func(*args, **kwargs)
  File "./web/vhosts/stats.example.com/htdocs/misc/log-analytics/import_logs.py", line 1586, in _call_api
    return json.loads(res)
  File "/usr/lib/python3.5/json/__init__.py", line 312, in loads
    s.__class__.__name__))
TypeError: the JSON object must be str, not 'bytes'
`
I tried the different proposals for the issue, without success.
I ended up finding this solution, which converts the JSON string well to the format wanted by json.loads at line 1586

It would indeed be interesting to try it on another environment than the one I used